samza-api/src/main/java/org/apache/samza/util/BlockingEnvelopeMap.java

```diff
@@ -144,6 +146,12 @@ private void initializeInternalStateForSSP(SystemStreamPartition systemStreamPar
           while (envelope == null && !isAtHead(systemStreamPartition)) {
             metrics.incBlockingPoll(systemStreamPartition);
             envelope = queue.poll(1000, TimeUnit.MILLISECONDS);
+
+            // Check for consumerFailure and throw exception
+            if (this.failureCause != null) {
```

Comment:
Minor: Can move before the metrics increment and poll (first thing in the loop).